Python/errors.c

```diff
@@ -151,7 +151,14 @@ _PyErr_SetObject(PyThreadState *tstate, PyObject *exception, PyObject *value)
     }
     Py_XINCREF(value);
     /* Normalize the exception */
-    if (value == NULL || (PyObject *)Py_TYPE(value) != exception) {
+    int is_subclass = 0;
+    if (value != NULL) {
+        is_subclass = PyObject_IsSubclass((PyObject*)Py_TYPE(value), exception);
```

Comment:
```suggestion
        is_subclass = PyObject_IsSubclass((PyObject *)Py_TYPE(value), exception);
```

Reply:
I don't think it can be, if it's NULL then it gets assigned fixed_value above. I can change this to be an assertion.